src/Raven.Server/Documents/Indexes/Static/StaticIndexBase.cs

```diff
@@ -437,6 +438,78 @@ protected IEnumerable<AbstractField> CreateField(string name, object value, bool
             });
         }
 
+        public unsafe dynamic AsDateOnly(dynamic field)
+        {
+            if (field is LazyStringValue lsv)
+            {
+                if (LazyStringParser.TryParseDateOnly(lsv.Buffer, lsv.Length, out var @do) == false) 
+                    return DynamicNullObject.Null;
+                
+                return @do;
+            }
+            
+            if (field is string str)
+            {
+                fixed (char* strBuffer = str.AsSpan())
+                {
+                    if (LazyStringParser.TryParseDateOnly(strBuffer, str.Length, out var to) == false)
+                        return DynamicNullObject.Null;
+
+                    return to;
+                }
+            }
+            
+            if (field is DateTime dt)
+                return DateOnly.FromDateTime(dt);
+
+            if (field is DateOnly dtO)
+                return dtO;
+
+            if (field is DynamicNullObject or null)
```

Comment:
1. If `filed` is `null` then we should return `DynamicNullObject.ExplicitNull`
2. If fields is already `DynamicNullObject` then just return it as is

Reply:
@ppekrol please have a look here as well

Reply:
Yes